### ISSUE TITLE
Unable to upload after following the documentation : 

### ISSUE THREAD

**Issue by tlylt** [ryanto/next-s3-upload#39]
Hi,
I tried following the docs for custom file input [here](https://next-s3-upload.codingvalue.com/custom-file-input)
but the file is not uploaded, and the following error is shown:
```
Unhandled Runtime Error
Error: Unsupported body payload object
```
![image](https://user-images.githubusercontent.com/41845017/150747371-0ac51945-bba2-446e-bb2f-7e75bcd81e1f.png)

**Comment by ryanto** [comment]
Oh wow that doc is wrong. I'll get it updated.

In the mean time use this...


```jsx
import { useState } from "react";
import { useS3Upload } from "next-s3-upload";

export default function UploadTest() {
  let [imageUrl, setImageUrl] = useState();
  let { uploadToS3 } = useS3Upload();

  let handleFileChange = async event => {
    let file = event.target.files[0];
    let { url } = await uploadToS3(file);

    setImageUrl(url);
  };

  return (
    <div>
      <input type="file" onChange={handleFileChange} />

      {imageUrl && <img src={imageUrl} />}
    </div>
  );
}
```

**Comment by tlylt** [comment]
> Ok, I updated the docs to show the right way to do it. Thanks for reporting this!
> 
> I need to get tests running against the examples in the docs :). I'm going to leave this issue opened until tests are setup. Please let me know if you run into any more issues.

Thanks for the quick response! I tried another example and it worked(Awesome guide btw)....Although I think there are some missing details in the AWS S3 setup. I followed it step by step and when trying to upload, I was faced with 403 Access Denied Errors and had to update the bucket policy to allow for put.

**Comment by ryanto** [comment]
Ah okay. I think when I setup the website I copied the old instructions! Opps!

I'll get the PUT fixed, did you run into anything else missing in the setup? Was anything confusing or hard to follow?

**Comment by ryanto** [comment]
Oh one more question... 

> had to update the bucket policy to allow for put.

Can you let me know what steps you had to do to get this fixed. I want to be sure I've got everything covered.

**Comment by tlylt** [comment]
> I'll get the PUT fixed, did you run into anything else missing in the setup? Was anything confusing or hard to follow?

Most instructions are quite clear and easy to follow 👍 

> Can you let me know what steps you had to do to get this fixed. I want to be sure I've got everything covered.

Can't remember the exact steps, but I first copied a policy from another tutorial somewhere, and I had to append `s3:PutObject*` to it. After adding this to the bucket policy, it started working (I don't think understand bucket policy enough to say that the following is safe to use, for anyone who's reading pls use it at your own risk).

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "PublicListGet",
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:List*",
                "s3:Get*",
                "s3:PutObject*"
            ],
            "Resource": [
                "arn:aws:s3:::amuze-storage",
                "arn:aws:s3:::amuze-storage/*"
            ]
        }
    ]
}
```

I remembered there was another issue. When examining the network requests, I saw that using the s3upload with the API setup as follows will make the filename undefined. It looks something like `localhost:3000/api/s3-upload/undefined`

```js
// pages/api/s3-upload.js
export { APIRoute as default } from 'next-s3-upload';
```

**Comment by ryanto** [comment]
Ah gotcha. Thanks for getting back!

As far as the policy goes I'd say it's best to use the one from the docs here: https://next-s3-upload.codingvalue.com/setup#iam-user. Can't make any promises about other policies.

I'll take a look at the API request, an `undefined` showing up there looks like a bug! I think it's possible to trigger an onChange with a null file, which could cause this. Might be as simple as another docs update :), but I'll write some tests in a few days and see what I can find.

Thanks again, I appreciate all the feedback, especially while it's fresh in your head.
